src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -260,6 +266,43 @@ public sealed partial class JsonNumber : System.Text.Json.JsonNode, System.IEqua
         [System.CLSCompliantAttribute(false)]
         public bool TryGetUInt64(out ulong value) { throw null; }
     }
+    public sealed partial class JsonObject : System.Text.Json.JsonNode, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>>, System.Collections.IEnumerable
+    {
+        public JsonObject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>> jsonProperties, System.Text.Json.DuplicatePropertyNameHandling duplicatePropertyNameHandling = System.Text.Json.DuplicatePropertyNameHandling.Replace) { }
+        public JsonObject(System.Text.Json.DuplicatePropertyNameHandling duplicatePropertyNameHandling = System.Text.Json.DuplicatePropertyNameHandling.Replace) { }
+        public System.Text.Json.JsonNode this[string key] { get { throw null; } set { } }
+        public System.Collections.Generic.ICollection<string> PropertyNames { get { throw null; } }
+        public System.Collections.Generic.ICollection<System.Text.Json.JsonNode> Values { get { throw null; } }
+        public void Add(System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode> jsonProperty) { }
+        public void Add(string propertyName, bool propertyValue) { }
```

Comment:
We should add support for Guid, DateTime, DateTimeOffset (also within JsonString).

Reply:
Wouldn't they be too specialized? I think @bartonjs had some thoughts on it.

Reply:
The existing JsonElement, JsonReader/Writer, support Guid, DateTime, DateTimeOffset. We should be consistent.
https://github.com/dotnet/corefx/blob/d9676859ae6b85b557fde919eeafd158629844cc/src/System.Text.Json/ref/System.Text.Json.cs#L64-L65
https://github.com/dotnet/corefx/blob/d9676859ae6b85b557fde919eeafd158629844cc/src/System.Text.Json/ref/System.Text.Json.cs#L546-L548

Reply:
I think it's fine to have accelerators for creating JsonString (or JsonObject value...values..) from those types, but I don't think we need to pollute JsonString with all of the getters.

The argument for JsonElement/Utf8JsonReader is that it can return those value types without allocating an intermediate object.  If you're reading via the loose/mutable DOM then you've already created the strings.  (If there's feedback later that they're needed then it's feedback-based; but given that the writable DOM is already a different shape than JsonElement I don't think that we should start with "every method from JsonElement needs a home in the JsonNode structure")

Reply:
> but I don't think we need to pollute JsonString with all of the getters.

What do you mean by this? I am not suggesting we add getters. I wanted us to be consistent in the types we support when creating a JsonObject.

For example, we have Add overloads on `JsonObject` that take property name and all the supported .NET numeric type. We could instead remove all of them and just have one:

```C#
Add(string propertName, JsonNode propertyValue);
```
And then, the user would need to create a JsonNumber from an byte, int, etc.

Since we decided to add convenience overloads for numeric types (which creates a `JsonNumber` for you), I would expect similar overloads for creating a `JsonString` from Guid, DateTime{Offset}, and similar overloads for `Add` that accept Guid, DateTime{Offset} and do the `JsonString` creation for me under the covers.

Reply:
> If you're reading via the loose/mutable DOM then you've already created the strings.

Don't make users have to convert DateTime/DateTimeOffset to string themselves. There should be overloads that convert between dates and ISO date string format.